backend/apps/ifc_validation/checks/header_policy/validate_header.py

```diff
@@ -90,36 +90,73 @@ class HeaderStructure(ConfiguredBaseModel):
     
     company_name: str = Field(default="")
     application_name: str = Field(default="")
+    schema_identifier: str = Field(default="")
     version: str = Field(default="")
+    mvd: str = Field(default="")
+    ifc_application_info: dict = Field(default_factory=dict)
     
     
     @model_validator(mode='before')
     def populate_header(cls, values):
         if (file := values.get('file')):
-            header = file.wrapped_data.header
+
+            purepythonparser = values.get('purepythonparser')
+            header = file.wrapped_data.header if not purepythonparser else file.header
             file_description = header.file_description
             file_name = header.file_name
             
+            fields = [
+                ('description', 0),
+                ('implementation_level', 1),
+                ('name', 0),
+                ('time_stamp', 1),
+                ('author', 2),
+                ('organization', 3),
+                ('preprocessor_version', 4),
+                ('originating_system', 5),
+                ('authorization', 6)
+            ]
+
             attributes = {
-                'description': file_description.description,
-                'implementation': file_description.implementation_level,
-                'name': file_name.name,
-                'time_stamp': file_name.time_stamp,
-                'author': file_name.author,
-                'organization': file_name.organization,
-                'preprocessor_version': file_name.preprocessor_version,
-                'originating_system': file_name.originating_system,
-                'authorization': file_name.authorization, 
-                'validation_errors' : []
+                field: getattr(file_description, field) if not purepythonparser else file_description[index]
+                for field, index in fields[:2]  # First two fields are equal to file_description
+            } | {
+                field: getattr(file_name, field) if not purepythonparser else file_name[index]
+                for field, index in fields[2:]  # Remaining fields are equal to file_name
             }
+            attributes['validation_errors'] = []
+            attributes['mvd'] = ''
+            attributes['schema_identifier'] = ''
             
             attributes = validate_and_split_originating_system(attributes)
```

Comment:
Oh yes, and is that "An error occurred" message for BC___.ifc solved?

Reply:
Yes :-) Running the file locally; 
![image](https://github.com/user-attachments/assets/69e04321-141a-42ea-b0ce-20f151ac9758)
https://github.com/buildingSMART/validate/pull/146/commits/ed20958fbbd1c01d9dcb90a4b145601aa2802bca

I've also added the file to the set of testfiles